src/libraries/Microsoft.Extensions.HostFactoryResolver/src/HostFactoryResolver.cs

```diff
@@ -341,15 +341,10 @@ public void OnNext(KeyValuePair<string, object?> value)
                     if (_stopApplication)
                     {
                         // Stop the host from running further
-                        throw new StopTheHostException();
+                        throw new HostAbortedException();
```

Comment:
@danmoseley Should we outsource this to a ThrowHelper class or does it not matter in that context?

Reply:
We use the throw helper pattern only in hot code, where it is worth it to make code shorter and perhaps more easily inlined. We do not do it in general. You should generally follow the pattern in other similar code.